File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int>setA,setB;
        vector<vector<int>>res(2);
        for(int i:nums1)setA.insert(i);
        for(int i:nums2)setB.insert(i);
        for(int i:setA)
            if(setB.find(i)==setB.end())res[0].push_back(i);
        for(int i:setB)
            if(setA.find(i)==setA.end())res[1].push_back(i);
        return res;
    }
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        List<List<Integer>>res=new ArrayList<>();
        List<Integer>l1=new ArrayList<>();
        List<Integer>l2=new ArrayList<>();
        Set<Integer>setA=new HashSet<Integer>();
        Set<Integer>setB=new HashSet<Integer>();
        for(int i:nums1)setA.add(i);
        for(int i:nums2)setB.add(i);
        
        for(int i:setA){
            if(setB.contains(i)==false)l1.add(i);
        }
        for(int i:setB){
            if(setA.contains(i)==false)l2.add(i);
        }
        res.add(l1);
        res.add(l2);
        return res;

    }
}